wire-ios-data-model/Source/Model/Message/ConversationMessage.swift

```diff
@@ -340,14 +342,37 @@ extension ZMMessage {
         return .delivered
     }
 
-    @objc public var usersReaction: [String: [UserType]] {
-        var result = [String: [ZMUser]]()
+    @objc public var reactionData: Set<ReactionData> {
+        var result = Set<ReactionData>()
         for reaction in reactions where reaction.users.count > 0 {
-            result[reaction.unicodeValue!] = [ZMUser](reaction.users)
+            result.insert(
+                ReactionData(
+                    reactionString: reaction.unicodeValue!,
+                    users: reaction.users.map { $0 },
+                    creationDate: reaction.creationDate
+                )
+            )
         }
         return result
     }
 
+    @objc public var usersReaction: [String: [UserType]] {
+        return reactionData
+            .map { $0 }
```

Comment:
Is `.map { $0 }` necessary?

Reply:
Yes, `Set` does not have partition method.

Reply:
Again, better to express this as `Array(reactionData)`.